nx/lib/nx.ex

```diff
@@ -17332,6 +17332,16 @@ defmodule Nx do
     axes = opts[:axes]
     keep_axes = opts[:keep_axes]
     max = reduce_max(tensor, axes: axes, keep_axes: true)
+
+    max =
+      case max do
+        %T{data: %Nx.Defn.Expr{}} = t ->
+          Nx.Defn.Kernel.stop_grad(t)
```

Comment:
@josevalim not sure if this should be done differently for compilation purposes

Reply:
I'm gonna merge, but feel free to suggest changes here. I can commit afterwards :)